src/main/java/graphql/DirectivesUtil.java

```diff
@@ -145,7 +144,14 @@ public DirectivesHolder(Collection<GraphQLDirective> allDirectives, Collection<G
 
             this.allAppliedDirectives = ImmutableList.copyOf(allAppliedDirectives);
             this.allAppliedDirectivesByName = ImmutableMap.copyOf(FpKit.groupingBy(allAppliedDirectives, GraphQLAppliedDirective::getName));
+        }
+
+        public static DirectivesHolder create(List<GraphQLDirective> directives, List<GraphQLAppliedDirective> appliedDirectives) {
+            if (directives.isEmpty() && appliedDirectives.isEmpty()) {
+                return EMPTY_HOLDER;
+            }
 
+            return new DirectivesHolder(directives, appliedDirectives);
```

Comment:
should we make the constructor private?

Reply:
I mean I think this is a private thing, but it's not marked with `@Internal` so I don't want to mess with the API.

Reply:
`@Deprecate` it then? For future removal?

Reply:
The `DirectivesUtil` class itself is marked with `@Internal`, maybe that means everything under it also is. Not sure...

Reply:
`@Internal` would apply to all methods inside the class

Reply:
Yup all code in the class is internal.  We can break it at will!  Use at your own risk!